jsonschema-core.xml

```diff
@@ -1476,87 +1508,221 @@
                         <xref target="RFC3986"> RFC 3986 Section 5.1.4</xref>.  It is RECOMMENDED
                         that implementations document any default base URI that they assume.
                     </t>
+                    <t>
+                        Unless the "$id" keyword described in the next section is present in the
+                        root schema, this base URI SHOULD be considered the canonical URI of the
+                        schema document's root schema resource.
+                    </t>
                 </section>
 
                 <section title='The "$id" Keyword' anchor="id-keyword">
                     <t>
-                        The "$id" keyword defines a URI for the schema, and the base URI that
-                        other URI references within the schema are resolved against.
-                        A subschema's "$id" is resolved against the base URI of its parent schema.
-                        If no parent schema defines an explicit base URI with "$id", the base URI
-                        is that of the entire document, as determined per
-                        <xref target="RFC3986">RFC 3986 section 5</xref>.
+                        The "$id" keyword identifies a schema resource with its
+                        <xref target="RFC6596">canonical</xref> URI.
+                    </t>
+                    <t>
+                        Note that this URI is an identifier and not necessarily a network locator.
+                        In the case of a network-addressable URL, a schema need not be downloadable
+                        from its canonical URI.
                     </t>
                     <t>
                         If present, the value for this keyword MUST be a string, and MUST represent a
-                        valid <xref target="RFC3986">URI-reference</xref>.
-                        This value SHOULD be normalized, and SHOULD NOT be an empty fragment &lt;#&gt;
-                        or an empty string &lt;&gt;.
+                        valid <xref target="RFC3986">URI-reference</xref>.  This URI-reference
+                        SHOULD be normalized, and MUST resolve to an
+                        <xref target="RFC3986">absolute-URI</xref> (without a fragment).  Therefore,
+                        "$id" MUST NOT contain a non-empty fragment, and SHOULD NOT contain an
+                        empty fragment.
```

Comment:
I've read and re-read this one paragraph a few times.
MUST NOT contain a non-empty fragment...
SHOULD NOT contain an empty fragment...

Aren't these inverse situations and so should have inverse requirement keywords?
I may just be not properly understanding what a "non-empty" fragment is. Not sure. Tired.

Reply:
Double negative. “MUST NOT contain a fragment other than #”

Reply:
Oooooooh I get it now. Morning rather than sleep time.
A non empty fragment would be `#something` and an empty fragment is `#`.
Allow empty fragment becaus legacy, but prefer no fragment at all.

Reply:
Looks like this was just me... =]

Reply:
OK I'm leaving this as-is.  "non-empty fragment" is phrasing that we have used in various places.